### DISCUSSION TITLE
Rustic-Browser

### DISCUSSION THREAD

**Discussion by durcom2** [rustic-rs/rustic#899] [Ideas]
A request for a Gui repository browser for Rustic the same as Restic-Browser that works with windows.
Maybe you can convince the developer to support Rustic as well as Restic as the repositories are the same format.

Restoring from commandline is a pain...

**Comment by simonsan** [comment]
Have you already tried opening a repository with the browser? What happens, if you do?

**Comment by durcom2** [comment]
Not sure what you are asking?  I have opened a **restic** repository with restic-browser...works fine...
Have not tried on a **Rustic** repository.  However, I want the browser to use Rustic and not restic for restores...mainly so I can use the Rustic "in place" restore and the restore --delete options...which restic does not have...and by extension, Restic-Browser also does not have since it uses Restic and not Rustic...

**Comment by aawsome** [comment]
1. Yes, the repository formats are compatible. rustic Adds some additional information, especially to the snapshots which can only be displayed if you use rustic to access the repository.
2. If you are talking about using the `rustic` binary instead or additionally to the `restic` binary within a tool like Restic-Browser, this should be very easy. But this is a feature request to that tool then, so please ask the corresponding maintainer(s) about this.
3. `rustic_core` as a library providing all needed functionality has just been released. So making a GUI without the need of additional binaries should be really easy. This is not top on the priority list of the `rustic` / `rustic_core` maintainers, but we can support anyone who wants to start such a project. Maybe even the guys from Restic-Browser could integrate `rustic_core`...

**Reply by durcom2** [reply]
Yes, I have also suggested to the restic-browser guys to support Rustic as well...

**Comment by durcom2** [comment]
restic-browser has released a version that allows you to specify the path to the "restic" executable to use.
https://github.com/emuell/restic-browser/issues/75
You can substitute the rustic executable instead of restic.
Haven't tested it yet, but it appears there is now a GUI restore for rustic under windows available.
